### PR TITLE
[MDS-4705] MineSpace Incident View and Submit Final Report

### PR DESCRIPTION
**Pull request by BenHarrington**
## Objective 

[MDS-4705](https://bcmines.atlassian.net/browse/MDS-4705)

_Changes_

- Ported over ScrollSideMenu component/styles from Core
- Implemented fixed side menu and header while navigation down the page(only when in final review state)
    - image
- Added status code callout widget to display when in a final review state(top of form)
    - image
- Added conditional rendering on IncidentPage/Form when in a final review state(not DFT, CLD, IRS)
    - Added Ministry Follow Up section(bottom of form) when in final review state
    - Added Documentation -> Final Report section when form determines it is a dangerous occurence(mine or EMLI)
    - 
image
- Added conditional validation on reported_to_phone_no/reported_to_email(one must be set at minimum)
- Created new modal for uploading incident documents/final report
- Removed old component files that were misnamed and abandoned

_How to Test_

- Visit the details page of an incident in a final review state(not DFT, CLD, IRS)
    - Verify:
        - All form inputs are disabled
        - Status code widget represents correct status
        - Ministry Follow Up section is displayed
        - Documentation -> Final Report section is displayed if incident is a dangerous occurrence
        - "Add Documentation/Add Final Report" buttons display next to document section titles which open an upload modal
            - Modal supports two different document types(INI, FIN) make sure both get uploaded with correct type(there is a known bug where uploading more documents to an existing incident mutates the document collection, ticket created)

### CODE REVIEW

File: services/minespace-web/src/components/Forms/incidents/IncidentForm.js
Comment (JustinMacaulay) [inline services/minespace-web/src/components/Forms/incidents/IncidentForm.js:154]: is there a reason there is a colour code inline here?

File: services/minespace-web/src/components/pages/Incidents/IncidentPage.js
Comment (JustinMacaulay) [inline services/minespace-web/src/components/pages/Incidents/IncidentPage.js]: could this block not just be an elvis operator in the className?

**Review by JustinMacaulay — COMMENTED** [review]
just a few comments

File: services/minespace-web/src/components/Forms/incidents/IncidentForm.js
Comment (BenHarrington) [reply services/minespace-web/src/components/Forms/incidents/IncidentForm.js:154]: This code was used to keep the styles similar with NoD which has this in it's own stylesheet. I didn't feel it was necessary to create a stylesheet for this component for one thing. If we continue to use this elsewhere(non Nod) we can build it into general styles. Unless you want that changed?

**Review by JustinMacaulay — DISMISSED** [review]
Looks good

**Review by JustinMacaulay — APPROVED** [review]
re-approving after other PR was merged in

**Pull request merged**
